### DISCUSSION TITLE
Masking URL address

### DISCUSSION THREAD

**Discussion by Camil88** [ThinkR-open/golem#741] [Q&A] [answered]
Hello, I'm using golem to write my app. For some reason I would like to mask localhost address and reaplace it with other one. So e.g. my address is `http://127.0.0.1:5475/` and I would like to mask it as `http://myappmadewithgolem.com`. Is it possible?

**Comment by ColinFay** [comment, marked as answer]
Hey, 

Thanks for using `{golem}`. 

This is not a question related to `{golem}` per se, but will concern any web application you want to serve to the world. 

Let me try and do a quick explanation. 

`127.0.0.1` refers to your __local computer__. This `127.0.0.1` is your computer when you run an app, and it will be my computer if I run the app on my computer. 

If you want to __make your application available to the world__, you'll need two things: 

+ A remote server, aka a computer on a given public IP address (for example 12.23.34.45) somewhere else that is running the application on a given port (for example port 1234). 
+ A domain name, aka a `myappmadewithgolem.com` URL that belongs to you

Once you have launched the application on the remote server, you can ask your domain provider (the company selling you the domain name) to point to your server. 
So basically you ask your domain provider "whenever someone type in their browser `myappmadewithgolem.com`, please show them the app available at `12.23.34.45:1234`.

This is not something very complex per se but requires a bit of IT knowledge. 

If everything I just said sounds alien to you, I would suggest going for a on premise service like shinyapps.io :) 

Hope this answers your question, 
Colin

**Comment by Camil88** [comment]
Hey Colin,

thanks for your answer. To be honest I'll probably serve my app on IIS server (I guess it's possible but seems quite hard, anyways I've heard it's less complex on Linux). The reason why I'm asking is that first I need to present my test app in front of some people, don't want them to see I'm running my app locally (may sound strange but I have my own reasons). That's why I was looking for a solution to somehow 'mask' my local address with some other. I was wondering if there's any possibility for doing so.

**Reply by ColinFay** [reply]
There is no way to do that with `{golem}` :/ 

Your browser __needs__ to know where to look for the app, and the app is on 127.0.0.1 :) 

The only way to mask an IP is to use an external DNS but still, that won't work on localhost. 

If you want them not to see that you are on 127.0.0.1, one solution would simply be to have your browser in full screen mode, so that the URL bar is not visible :)

**Reply by Camil88** [reply]
Ok, thanks for your answer, I thought it could be possible with `golem` itself. Your suggestions are great, thanks. Actually, I did some trick - I opened my app in the browser and then installed the page as an application (on Edge it's Settings>Applications>Install as application). It means I can create shortcut of web app on a desktop. Once I double click app icon on desktop then  it opens up in a browser and no address bar is visible, it's hidden by default. So it worked for me, anyway thank you for discussing this topic with me here and your kind replies.
